src/hooks/useActiveOnboard.ts

```diff
@@ -14,125 +9,117 @@ import web3Utils from 'web3-utils'
 
 import { SupportedChainId, UnsupportedChainId } from '@gooddollar/web3sdk'
 
-import { 
-  useConnectWallet,
-  useSetChain,
-  useWallets
-} from '@web3-onboard/react'
+import { useConnectWallet, useSetChain, useWallets } from '@web3-onboard/react'
 
 export type IsSupportedChainId = {
-  isSupported: boolean,
-  chainId: string
+    isSupported: boolean
+    chainId: string
 }
 
 type OnboardConnectProps = {
-  activated: boolean,
-  tried: boolean
+    activated: boolean
+    tried: boolean
 }
 
-export const WalletLabels: Readonly<string[]> = ["WalletConnect", "ZenGo", "Coinbase Wallet"]
+export const WalletLabels: Readonly<string[]> = ['WalletConnect', 'ZenGo', 'Coinbase Wallet']
 export const WalletLinkKeys: Readonly<string[]> = [
-  '-walletlink:https://www.walletlink.org:Addresses',
-  '-walletlink:https://www.walletlink.org:session:secret',
-  '-walletlink:https://www.walletlink.org:session:id'
+    '-walletlink:https://www.walletlink.org:Addresses',
+    '-walletlink:https://www.walletlink.org:session:secret',
+    '-walletlink:https://www.walletlink.org:session:id',
 ]
 
-export type ActiveOnboard<T= any> = Omit<Web3ReactContextInterface<Web3Provider>, 'activate' | 'deactivate' | 'setError' | 'connector'>
+export type ActiveOnboard<T = any> = Omit<
+    Web3ReactContextInterface<Web3Provider>,
+    'activate' | 'deactivate' | 'setError' | 'connector'
+>
 
 export interface EIP1193ProviderExtended extends EIP1193Provider {
-  providers?: any
-  isMetamask?: boolean,
-  selectedProvider: {
-    isMetaMask?: boolean, 
-    on?: (...args: any[]) => void,
-    off?: (...args: any[]) => void,
-    removeListener?: (...args: any[]) => void,
-    removeAllListeners?: (...args: any[]) => void,
-    autoRefreshOnNetworkChange?: boolean,
-    request?: (args: {
-      method: string;
-      params?: unknown[] | object;
-  }) => Promise<unknown>,
-  } | null
+    providers?: any
+    isMetamask?: boolean
+    selectedProvider: {
+        isMetaMask?: boolean
+        on?: (...args: any[]) => void
```

Comment:
1. always declare method as methods, not as the props with function value
2. declare nested object as separate interface (see comment above)
3. use Partial generic to make all the following props optional

```suggestion
        on?: (...args: any[]) => void
```

Reply:
e.g.

```typescript
interface IProvider {
  isMetaMask?: boolean;
  on(...args: any[]): void;
  // ...
}

// ...
isMetamask?: boolean;
selectedProvider?:  Partial<IProvider>;
```